tests/infra.py

```diff
@@ -22,6 +22,21 @@ def get_env(varname):
     return os.environ[varname]
 
 
+class ExpectedErrorFields(typing.NamedTuple):
+    code: str
+
+    status: typing.Optional[int] = None
+    """
+    If this is None, then we not check the status.  For all other values, we test that it matches.
+    """
+
+    expect_stacktrace: typing.Optional[bool] = None
+    """
+    If this is True, then we expect the stacktrace to be present.  If this is False, then we expect the stacktrace to be
+    absent.  If this is None, then we do not test the presence of the stacktrace.
+    """
```

Comment:
I don't know.  Python docstrings seem to follow the definition, unlike Java.  I followed that convention.

https://www.python.org/dev/peps/pep-0224/

this was rejected but it does put the comment below.